data/modules/FlightLog/FlightLogExporter.lua

```diff
@@ -0,0 +1,137 @@
+-- Copyright © 2008-2023 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local FlightLog = require 'modules.FlightLog.FlightLog'
+local FileSystem = require 'FileSystem'
+local Character = require 'Character'
+local Lang = require 'Lang'
+local l = Lang.GetResource("ui-core")
+
+local text_formatter = {}
+
+function text_formatter:open( file )
+	self.file = file
+	return self
+end
+
+function text_formatter:write( string )
+	self.file:write( string )
+	return self
+end
+
+function text_formatter:newline()
+	self.file:write( "\n" )
+	return self
+end
+
+function text_formatter:close()
+	self.file:close()
+end
```

Comment:
Remember to assign `self.file = nil` to avoid keeping a dangling reference to the closed file handle in the Lua GC - otherwise it will remain reachable until the next time the player exports a log.